extensions/toggl-track/src/toggl/index.ts

```diff
@@ -74,9 +76,9 @@ const TogglAPI = function (apiToken: string) {
 
 interface Preferences {
   togglApiToken: string;
+  hideArchivedProjects: boolean;
 }
 
-const preferences: Preferences = getPreferenceValues();
-const toggl = TogglAPI(preferences.togglApiToken);
+const toggl = TogglAPI();
```

Comment:
I suggest taking in { togglApiToken, hideArchivedProjects } as params to TogglAPI as before, now config is a side-effect within TogglAPI

Reply:
Other than that great work!

Reply:
I'm cool with that. Is there any specific benefit, or is that just good practice?